charts/hivemq-platform/templates/hivemq-custom-resource.yml

```diff
@@ -233,19 +242,20 @@ spec:
             targetPort: {{ include "hivemq-platform.range-service-port-name" . | trimAll "-" | trunc 15 | trimSuffix "-" | trim }}
           {{- if $val.port}}
             port: {{ $val.port }}
-          {{- else}}
+          {{- else }}
             port: {{ $val.containerPort }}
           {{- end }}
     {{- end }}
     {{- end }}
-    # Metric service ports are not configurable right now
+    {{- if and (not $hasMetricService) (.Values.metrics.enabled) }}
     - metadata:
-        name: {{ (include "hivemq-platform.range-service-name" (dict "releaseName" $.Release.Name "type" "metrics" "port" $defaultMetricsPort "containerPort" $defaultMetricsPort)) | trimAll "-" | trunc 63 | trimSuffix "-" |  trim }}
+        name: {{ (include "hivemq-platform.range-service-name" (dict "name" "" "releaseName" $.Release.Name "type" "metrics" "port" $metricsPort "containerPort" $metricsPort)) }}
       spec:
         ports:
-          - name: {{ printf "metrics-%s" $defaultMetricsPort }}
-            targetPort: {{ printf "metrics-%s" $defaultMetricsPort }}
-            port: {{ printf "%d" (int64 $defaultMetricsPort) }}
+          - name: {{ printf "metrics-%s" $metricsPort }}
+            targetPort: {{ printf "metrics-%s" $metricsPort }}
```

Comment:
Do we officially need to add `"metrics"` as supported type to `hivemq-platform.range-service-port-name`? I assume right now this is working since we fallback to `{{- printf "%s-%s" .type (toString .containerPort) }}` in the helper method?

Maybe adding `"metrics`" to the helper method and using it here would make it easier to guarantee that the `StatefulSet` and services configuration are always in sync.

Reply:
Changed the helper method as agreed already